### PR TITLE
Litmusctl End-to-End Tests

### PR DESCRIPTION
**Pull request by shivam-Purohit**
## Proposed changes

Summarize your changes here to communicate with the reviewers and make sure to put the link of the related issue

## How has this been tested:
- [ ] Test environment
    * [ ] GKE
    * [ ] AWS
    * [ ] OpenShift
    * [ ] KinD/k3s/k3d cluster
    * [ ] Others
 - [ ] Tested with production environment ( For ChaosCenter Tests )
 - [ ] Have not tested

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus-e2e/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added Screenshots or Terminal snippets in the PR comment to validate the successful execution of changes

**Documentation**
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] data-cy=* parameters have been added in codebase ( ChaosCenter Tests )
- [ ] data-cy=* parameters need to be added in codebase ( ChaosCenter Tests )
- [ ] No updates required.

**Code Review**
- [ ] Does the test handle fatal exceptions, ie. uncaught exceptions from 3rd party library (if any)

**Issue**
- [ ] Tasks in issue are checked off

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

File: litmus/litmusctl_tests.sh [hunk 1]
```diff
@@ -1,64 +1,80 @@
 #!/bin/bash
 
+set -o pipefail
+
 # All intermediate functions are defined in utils.sh
 source litmus/utils.sh
 
 namespace=${AGENT_NAMESPACE}
+# namespace='shivamtestnamespace'
```
Comment (Jonsy13) [inline litmus/litmusctl_tests.sh:9]: Can we remove commented lines

File: litmus/litmusctl_tests.sh [hunk 2]
```diff
@@ -70,71 +86,151 @@ function wait_for_agent_to_be_ready(){
...
+    infra_cleanup
+    delete_environment $envName
 }
 
 function test_get_projects(){
     configure_account
 
-    projects=$(litmusctl get projects | wc -l)
+    projects=$( echo "q" | litmusctl get projects | wc -l)
+    echo "${projects}"
```
Comment (Jonsy13) [inline litmus/litmusctl_tests.sh:202]: we can remove this `echo`

File: litmus/litmusctl_tests.sh [hunk 3]
```diff
@@ -143,40 +239,266 @@ function test_get_agents(){
...
-    disconnect_agent chaos_delegate_id projectID
+    disconnect_infra $chaos_infra_id $projectID
 
-    chaos_delegate_id=$(litmusctl get chaos-delegates --project-id=$projectID | grep "${agentName}" | awk '{print $1}')
+    chaos_infra_id=$(litmusctl get chaos-infra --project-id=$projectID | grep "${infraName}" | awk '{print $1}')
+    
+    infra_cleanup
+    delete_environment $envName    
+    if [[ ${chaos_infra_id} == "" ]];then
+        echo -e "\n[Info]: litmusctl disconnect chaos-infra working fine ✓\n"
```
Comment (Jonsy13) [inline litmus/litmusctl_tests.sh:271]: add `exit 0` here as well

File: litmus/litmusctl_tests.sh [hunk 4]
```diff
@@ -143,40 +239,266 @@ function test_get_agents(){
...
+
+function test_save_experiment(){
+    configure_account
+    printf "\n account created successfully"
+    projectID=$(echo "q" | litmusctl get projects | grep "$projectName" | awk '{print $1}')
+    printf "\n projectID is ${projectID}"
+    echo "E" | sudo add-apt-repository ppa:rmescandon/yq
+    sudo apt-get install yq 
+    echo | echo "q" | litmusctl get chaos-experiments --project-id=$projectID --output="table"
+    yq --version
```
Comment (Jonsy13) [inline litmus/litmusctl_tests.sh:345]: we can remove this command

File: litmus/litmusctl_tests.sh [hunk 5]
```diff
@@ -143,40 +239,266 @@ function test_get_agents(){
...
+function test_delete_experiment(){
+    configure_account
+    printf "\naccount created successfully"
+    # capturing project id
+    projectID=$(echo "q" | litmusctl get projects | grep "${projectName}" | awk '{print $1}' )
+    printf "\nprojectID is ${projectID}"
+
+    # create before deleting
+    save_experiment
+    printf "\nexperiment id should have been ${expName}"
```
Comment (Jonsy13) [inline litmus/litmusctl_tests.sh:424]: we can remove this print

File: litmus/litmusctl_tests.sh (hunk 4 above)
Comment (shivam-Purohit) [reply litmus/litmusctl_tests.sh]: thas nice

**Review by Jonsy13 — APPROVED** [review]

**Review by Saranya-jena — APPROVED** [review]

**Comment by Jonsy13** [comment]
Thanks @shivam-Purohit for adding tests for `litmusctl`. This is going to help a lot.
Merging this PR now as changes are looking good!

**Comment by shivam-Purohit** [comment]
> Thanks @shivam-Purohit for adding tests for `litmusctl`. This is going to help a lot. Merging this PR now as changes are looking good!

not possible without your help.

**Pull request merged**